extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesPeonLifecycle.java

```diff
@@ -260,11 +261,19 @@ protected TaskLocation getTaskLocation()
     since Druid doesn't support retrying tasks from a external system (K8s). We can explore adding a fabric8 watcher
     if we decide we need to change this later.
     **/
-    if (taskLocation == null) {
+    if (!taskLocation.isDone()) {
       log.warn("Unknown task location for [%s]", taskId);
```

Comment:
i'm a little unsure about this logic because i have seen instances where the k8s calls can temporarily fail. i am also not sure if fabric8 is actually retrying these calls.

see the initial trigger for (https://github.com/apache/druid/pull/17431, https://github.com/apache/druid/pull/17417)

i think i would feel better if we still fell back to querying k8s for the pod location in this block for now (returning a immediate future). i think another option would be to wrap getTaskLocationFromK8s in more retries, i can put up a separate pr if you don't want to do it in this one though.

Reply:
What change are you suggesting here? I can make it, but I'm not sure what you mean exactly.

Reply:
basically i think for now we should add a check for the future having completed with a exception and retry it

Reply:
I think we're deciding to address this through #17417 instead. Most of what's happening in this PR is about making it so locations are always known by the time `start()` exits, whereas #17417 is about making that unnecessary by fetching locations live.